gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/DomainServiceImpl.java

```diff
@@ -435,6 +435,11 @@ public Single<Domain> update(String domainId, Domain domain) {
         return domainRepository.findById(domainId)
                 .switchIfEmpty(Single.error(new DomainNotFoundException(domainId)))
                 .flatMap(existingDomain -> {
+                    if(domain.getDataPlaneId() != null){
+                        if(!domain.getDataPlaneId().equals(existingDomain.getId())){
```

Comment:
```suggestion
                        if(!domain.getDataPlaneId().equals(existingDomain.getDataPlaneId())){
```
TODO: can you please add a test case for this one ?